File: Python/Algorithms/1237.py
class Solution:
    def findSolution(self, customfunction, z):
        """ """
        # Annoying how i, j supposedly valid in range(1, 1001) but
        # can completely bust down # of iterations when noticing how
        # they all take values [0, 100]
        func, res, range_ = customfunction.f, [], range(1, 101)
        append = res.append
        for i in range_:
            for j in range_:
                result = func(i, j)
                if result == z:
                    append([i, j])
                    break
                if result > z:
                    break
        return res

File: Rust/Algorithms/1237.rs
/*
 * // This is the custom function interface.
 * // You should not implement it, or speculate about its implementation
 * struct CustomFunction;
 * impl CustomFunction {
 *    pub fn f(x:i32,y:i32)->i32{}
 * }
 */

impl Solution {
    pub fn find_solution(customfunction: &CustomFunction, z: i32) -> Vec<Vec<i32>> {
        let mut res = Vec::new();
        for i in 1..101 {
            for j in 1..101 {
                let result = customfunction.f(i, j);
                if result == z {
                    res.push(vec![i, j]);
                    break;
                }
                if result > z {
                    break;
                }
            }
        }
        res
    }
}
